File: Backend/Microservices/Commodity/Eefa.Commodity.Application/Commands/CommodityCategoryPropertyItem/Create/CreateCommodityCategoryPropertyItemCommand.cs
using AutoMapper;
using Eefa.Common.CommandQuery;
using Eefa.Common;
using Eefa.Common.Data;
using MediatR;
using System.Threading;
using System.Threading.Tasks;


namespace Eefa.Commodity.Application.Commands.CommodityCategoryPropertyItem.Create
{
    public class CreateCommodityCategoryPropertyItemCommand : CommandBase, IRequest<ServiceResult>, IMapFrom<CreateCommodityCategoryPropertyItemCommand>, ICommand
    {

        /// <summary>
        /// کد ویژگی گروه
        /// </summary>
        public int CategoryPropertyId { get; set; } = default!;

        /// <summary>
        /// کد والد
        /// </summary>
        public int? ParentId { get; set; }

        /// <summary>
        /// عنوان
        /// </summary>
        public string Title { get; set; } = default!;

        /// <summary>
        /// نام اختصاصی
        /// </summary>
        public string UniqueName { get; set; } = default!;

        /// <summary>
        /// کد
        /// </summary>
        public string? Code { get; set; }

        /// <summary>
        /// ترتیب نمایش
        /// </summary>
        public int OrderIndex { get; set; } = default!;

        /// <summary>
        /// فعال است؟
        /// </summary>
        public bool IsActive { get; set; } = default!;

        public void Mapping(Profile profile)
        {
            profile.CreateMap<CreateCommodityCategoryPropertyItemCommand, Data.Entities.CommodityCategoryPropertyItem>();
        }
    }

    public class CreateCommodityCategoryPropertyItemCommandHandler : IRequestHandler<CreateCommodityCategoryPropertyItemCommand, ServiceResult>
    {
        private readonly IRepository<Data.Entities.CommodityCategoryPropertyItem> _repository;
        private readonly IMapper _mapper;

        public CreateCommodityCategoryPropertyItemCommandHandler(IRepository<Data.Entities.CommodityCategoryPropertyItem> repository, IMapper mapper)
        {
            _mapper = mapper;
            _repository = repository;
        }


        public async Task<ServiceResult> Handle(CreateCommodityCategoryPropertyItemCommand request, CancellationToken cancellationToken)
        {
            var commodityCategoryPropertyItem = _mapper.Map<Data.Entities.CommodityCategoryPropertyItem>(request);

            var entity = _repository.Insert(commodityCategoryPropertyItem);

            await _repository.SaveChangesAsync(cancellationToken);

            return ServiceResult.Success();
        }
    }
}

File: Frontend/angular-app/src/app/modules/commodity/repositories/commodity-category-property-item/commands/create-commodity-category-property-item-command.ts
import {Inject} from "@angular/core";
import {ServiceResult} from "../../../../../core/models/service-result";
import {IRequest, IRequestHandler} from "../../../../../core/services/mediator/interfaces";
import {HttpService} from "../../../../../core/services/http/http.service";
import {ApplicationError} from "../../../../../core/exceptions/application-error";
import {MediatorHandler} from "../../../../../core/services/mediator/decorator";
import {NotificationService} from "../../../../../shared/services/notification/notification.service";
import {CommodityCategoryPropertyItem} from "../../../entities/commodity-category-property-item";
import {ValidationRule} from "../../../../../core/validation/validation-rule";
import {HttpRequest} from "../../../../../core/services/http/http-request";
import {EntityStates} from "../../../../../core/enums/entity-states";

export class CreateCommodityCategoryPropertyItemCommand extends IRequest<CreateCommodityCategoryPropertyItemCommand, CommodityCategoryPropertyItem> {
  public state: number = EntityStates.Inserted;
  public categoryPropertyId: number | undefined = undefined;
  public parentId: number | undefined = undefined;
  public title: string | undefined = undefined;
  public uniqueName: string | undefined = undefined;
  public code: string | undefined = undefined;
  public orderIndex: number | undefined = undefined;
  public isActive: boolean | undefined = undefined;

  constructor() {
    super();
  }

  mapFrom(entity: CommodityCategoryPropertyItem): CreateCommodityCategoryPropertyItemCommand {
    throw new ApplicationError(CreateCommodityCategoryPropertyItemCommand.name, this.mapTo.name, 'Not Implemented Yet')
  }

  mapTo(): CommodityCategoryPropertyItem {
    throw new ApplicationError(CreateCommodityCategoryPropertyItemCommand.name, this.mapTo.name, 'Not Implemented Yet')
  }

  get url(): string {
    return "/commodity/commodityCategoryPropertyItem/add";
  }

  get validationRules(): ValidationRule[] {
    return [];
  }
}

@MediatorHandler(CreateCommodityCategoryPropertyItemCommandHandler.name)
export class CreateCommodityCategoryPropertyItemCommandHandler implements IRequestHandler<CreateCommodityCategoryPropertyItemCommand, CommodityCategoryPropertyItem> {
  constructor(
    @Inject(HttpService) private _httpService: HttpService,
    @Inject(NotificationService) private _notificationService: NotificationService,
  ) {
  }

  async Handle(request: CreateCommodityCategoryPropertyItemCommand): Promise<CommodityCategoryPropertyItem> {
    let httpRequest: HttpRequest<CreateCommodityCategoryPropertyItemCommand> = new HttpRequest<CreateCommodityCategoryPropertyItemCommand>(request.url, request);
    // httpRequest.Query  += ;
    // httpRequest.Headers = httpRequest.Headers.append()

    return await this._httpService.Post<CreateCommodityCategoryPropertyItemCommand, ServiceResult<CommodityCategoryPropertyItem>>(httpRequest).toPromise().then(response => {
      // this._notificationService.showHttpSuccessMessage()
      return response.objResult
    })
  }
}
